bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/slider/AjaxBootstrapSlider.java

```diff
@@ -0,0 +1,67 @@
+package de.agilecoders.wicket.extensions.slider;
+
+import org.apache.wicket.ajax.AbstractDefaultAjaxBehavior;
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.request.cycle.RequestCycle;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * AjaxBootstrapSlider
+ */
+public class AjaxBootstrapSlider<T extends ISliderValue, N extends Number> extends BootstrapSlider<T,N> {
+
+    private AbstractDefaultAjaxBehavior behavior;
+    
+    public static enum SliderEvent {
+        slide,	//This event fires when the slider is dragged        
+        slideStart,	//This event fires when dragging starts	The new slider value
+        slideStop,	//This event fires when the dragging stops or has been clicked on	The new slider value
+        change 
+    }
+    
+    public static interface EventHandler<T extends ISliderValue> extends Serializable {
+        
+        public void onAjaxEvent(AjaxRequestTarget target, T newValue);
+    }
+    
+    private Map<SliderEvent, EventHandler> handlers = new HashMap<SliderEvent, EventHandler>();
+    
+    public AjaxBootstrapSlider(String id, IModel<T> model, Class<T> typeClass) {
+        super(id, model, typeClass);
+
+        behavior = new AbstractDefaultAjaxBehavior() {
+            @Override
+            protected void respond(AjaxRequestTarget target) {
+                SliderEvent event = SliderEvent.valueOf(RequestCycle.get().getRequest().getRequestParameters().getParameterValue("event").toString());
```

Comment:
Cache the RequestParameters since it is looked up more than once in this method.